src/values/AbstractValue.js

```diff
@@ -666,6 +666,9 @@ export default class AbstractValue extends Value {
     isCondition?: boolean,
     doNotSimplify?: boolean
   ): Value {
+    if (left === right) {
+      return left;
```

Comment:
If both are `undefined ` then return value would be `undefined` too?
I think maybe you want `left || realm.intrinsics.undefined`, like below.